diffrax/_term.py

```diff
@@ -776,3 +785,266 @@ def _to_vjp(_y, _diff_args, _diff_term):
         dy, vjp = jax.vjp(_to_vjp, y, diff_args, diff_term)
         da_y, da_diff_args, da_diff_term = vjp((-(a_y**ω)).ω)
         return dy, da_y, da_diff_args, da_diff_term
+
+
+# The Underdamped Langevin SDE trajectory consists of two components: the position
+# `x` and the velocity `v`. Both of these have the same shape.
+# So, by UnderdampedLangevinX we denote the shape of the x component, and by
+# UnderdampedLangevinTuple we denote the shape of the tuple (x, v).
+UnderdampedLangevinLeaf = Shaped[Array, "*underdamped_langevin"]
+UnderdampedLangevinX = PyTree[
+    Shaped[Array, "?*underdamped_langevin"], "UnderdampedLangevinX"
+]
+UnderdampedLangevinTuple = tuple[UnderdampedLangevinX, UnderdampedLangevinX]
+
+
+class UnderdampedLangevinStructureError(Exception):
```

Comment:
I would like your opinion on this Patrick. I realised that if `gamma` or `u` are indeed shaped incorrectly, then Diffrax throws the following:
```python
ValueError: `terms` must be a PyTree of `AbstractTerms` (such as `ODETerm`), with structure diffrax._term.MultiTerm[tuple[diffrax._term.AbstractTerm[typing.Any, typing.Union[jaxtyping.Float[Array, ''], jaxtyping.Float[ndarray, ''], numpy.number, float, jaxtyping.Int[Array, ''], jaxtyping.Int[ndarray, ''], int]], diffrax._term.AbstractTerm[typing.Any, diffrax._custom_types.AbstractSpaceTimeLevyArea]]]
```

This comes from `_integrate._term_compatible` and is rather unhelpful. But if we want a different error message, then we must throw an exception that is not a `ValueError`, because those are caught by `_term_compatible`. Do you think this solution is appropriate, or do you suggest something else?

Also because of this I kept `_broadcast_pytree` in here, and then the function `broadcast_underdamped_langevin_arg` defined below gets imported by `foster_langevin_srk.py`. Is that okay?

Reply:
Switching that to a different error type sound reasonable. .Although it shouldn't matter too much, the original error should still be visible in the stack trace above.

Rather than creating a new error type though, I think let's just make this one a `RuntimeError` for simplicity.

The import seems fine to me.

Reply:
Good, I'll swap it for a `RuntimeError` then.